iree/compiler/Dialect/LinalgPlus/IR/LinalgPlusOps.td

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_DIALECT_LINALGPLUS_OPS
+#define IREE_DIALECT_LINALGPLUS_OPS
+
+include "iree/compiler/Dialect/LinalgPlus/IR/LinalgPlusBase.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "mlir/Interfaces/ControlFlowInterfaces.td"
+
+//===----------------------------------------------------------------------===//
+// Base class.
+//===----------------------------------------------------------------------===//
+
+class LinalgPlus_Op<string mnemonic, list<OpTrait> traits = []> :
+    Op<LinalgPlus_Dialect, mnemonic, traits> {
+  let verifier = [{ return verify$cppClass(*this); }];
+}
+
+//===----------------------------------------------------------------------===//
+// Non-structured ops
+//===----------------------------------------------------------------------===//
+
+def LinalgPlus_SortOp : LinalgPlus_Op<"sort", [RecursiveSideEffects,
+                                               SameOperandsAndResultShape]> {
+  let summary = "Sort operator";
+  let description = [{
+    Based on XLA operation semantics, sorts the given `operands` at the given
+    `dimension` with the given `comparator`.
+
+    See https://www.tensorflow.org/xla/operation_semantics#sort.
+  }];
+  let arguments = (ins
+    Variadic<AnyType>:$operands,
```

Comment:
Actually why is this variadic. Should be just one operand right?

I was looking to MLIR `OpBase.td` . It might be easier to just define a new Tablegen type that checks for ranked tensor or ranked memref type. Something like `AnyRankedShapedType`. It doesnt exist in OpBase.td (might be worth adding there directly)

Reply:
Ok. i see this is to sort multiple operands at once. Thats fine maybe.

Reply:
`mhlo.sort` can take multiple operands at once, this is following the semantics.